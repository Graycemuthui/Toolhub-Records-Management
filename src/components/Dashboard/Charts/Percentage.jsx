import React from "react";
import { Pie } from "react-chartjs-2";
import { records } from "../../data/Mockdata";
import checkPercentage from "../../data/Mockdata";
import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";
ChartJS.register(ArcElement, Tooltip, Legend);

let percentageData = [checkPercentage(records), 100 - checkPercentage(records)];
const data = {
  labels: ["Edited Records", "Total Records"],
  datasets: [
    {
      label: "My First dataset",
      backgroundColor: ["rgb(255, 99, 132)", "rgb(0,0,255)"],
      borderColor: "rgb(0,0,255)",
      data: percentageData,
    },
  ],
};

function Percentage() {
  return <Pie data={data} />;
}

export default Percentage;
