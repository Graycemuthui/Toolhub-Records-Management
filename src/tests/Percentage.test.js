import React from "react";
import TestRenderer from "react-test-renderer";
import Percentage from "../components/Dashboard/Charts/Percentage";

it("Percentage chart render correctly", () => {
  const tree = TestRenderer.create(<Percentage />).toJSON();
  expect(tree).toMatchSnapshot();
});
