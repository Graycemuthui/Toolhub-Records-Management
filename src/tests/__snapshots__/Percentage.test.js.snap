// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Percentage chart render correctly 1`] = `
<canvas
  height={150}
  role="img"
  width={300}
/>
`;
